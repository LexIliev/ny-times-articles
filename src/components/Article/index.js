import React from 'react';
import PropTypes from 'prop-types';
import { ROUTES } from '../../routes/constants';
import { getRoute, checkEmpty } from '../../utils';

import './styles.scss';

const Article = ({
  id,
  title,
  byline,
  published_date,
  media,
  icon,
  history,
  handleClick,
}) => (
  <article className="article" onClick={handleClick}>
    <img
      className="article__image"
      src={
        media[0] &&
        media[0]['media-metadata'] &&
        media[0]['media-metadata'][0].url
      }
      alt={title}
    />
    <div className="article__wrapper">
      <h3 className="article__title">{title !== '' ? title : ' - '}</h3>
      <div className="article__info-wrapper">
        <p className="article__byline">{byline !== '' ? byline : 'By -'}</p>
        <p className="article__published-date">
          {published_date !== '' ? published_date : 'N/A'}
        </p>
      </div>
    </div>
    <img className="article__icon" src={icon} alt="Arrow Icon" />
  </article>
);

Article.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  byline: PropTypes.string.isRequired,
  published_date: PropTypes.string.isRequired,
  media: PropTypes.array.isRequired,
  icon: PropTypes.any.isRequired,
  handleClick: PropTypes.func,
};

export default Article;
