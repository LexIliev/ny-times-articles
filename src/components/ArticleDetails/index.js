import React from 'react';
import PropTypes from 'prop-types';
import { ROUTES } from '../../routes/constants';

import './styles.scss';

const ArticleDetails = ({
  id,
  title,
  abstract,
  byline,
  published_date,
  media,
}) => (
  <div className="article-details">
    <h3 className="article-details__title">{title}</h3>
    <p className="article-details__abstract">{abstract}</p>
    <p className="article-details__byline">{byline}</p>
    <p className="article-details__published-date">{published_date}</p>
    <img
      className="article-details__image"
      src={
        media[0] &&
        media[0]['media-metadata'] &&
        media[0]['media-metadata'][2].url
      }
      alt={title}
    />
  </div>
);

ArticleDetails.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  abstract: PropTypes.string.isRequired,
  byline: PropTypes.string.isRequired,
  published_date: PropTypes.string.isRequired,
  media: PropTypes.array.isRequired,
};

export default ArticleDetails;
