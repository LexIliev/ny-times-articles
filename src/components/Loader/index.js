import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const Loader = ({ loaderText }) => (
  <div>
    <h1 className="text">{loaderText}</h1>
  </div>
);

Loader.defaultProps = {
  loaderText: 'Loading...',
};

Loader.propTypes = {
  loaderText: PropTypes.string,
};

export default Loader;
