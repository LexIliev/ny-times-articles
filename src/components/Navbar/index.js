import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const menuIcon = require('../../../assets/three-horizontal-lines.svg');
const searchIcon = require('../../../assets/search.svg');
const dotsIcon = require('../../../assets/dots.svg');

const Navbar = ({ navigationTitle, menuIcon, searchIcon, dotsIcon }) => (
  <header className="navbar">
    <img
      className="navbar__menu"
      src={menuIcon}
      alt="Tree Horizontal Lines Icon"
    />
    <h2 className="navbar__title">{navigationTitle}</h2>
    <div className="navbar__icons-wrapper">
      <img className="navbar__item" src={searchIcon} alt="Search Icon" />
      <img className="navbar__item" src={dotsIcon} alt="Dots Icon" />
    </div>
  </header>
);

Navbar.defaultProps = {
  menuIcon: menuIcon,
  searchIcon: searchIcon,
  dotsIcon: dotsIcon,
};

Navbar.propTypes = {
  navigationTitle: PropTypes.string.isRequired,
  menuIcon: PropTypes.any.isRequired,
  searchIcon: PropTypes.any.isRequired,
  dotsIcon: PropTypes.any.isRequired,
};

export default Navbar;
