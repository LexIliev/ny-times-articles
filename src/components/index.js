import Article from './Article';
import ArticleDetails from './ArticleDetails';
import Navbar from './Navbar';
import Loader from './Loader';

export { Article, ArticleDetails, Navbar, Loader };
