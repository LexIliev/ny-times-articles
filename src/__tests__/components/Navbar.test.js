import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { create } from 'react-test-renderer';
import { Navbar } from '../../components';

describe('Navbar component', () => {
  jest.mock('./styles.scss', () => ({}));
  const mockedMenuIcon = '/assets/three-horizontal-lines.svg';
  const mockedSearchIcon = '/assets/search.svg';
  const mockedDotsIcon = '/assets/dots.svg';

  let props = {};

  beforeEach(() => {
    jest.mock('./styles.scss', () => ({}));
    props = {
      navigationTitle: '',
      menuIcon: mockedMenuIcon,
      searchIcon: mockedSearchIcon,
      dotsIcon: mockedDotsIcon,
    };
  });

  test('should match snapshot and render correctly', () => {
    props = {
      ...props,
      navigationTitle: 'Navigation title',
    };

    const component = create(<Navbar {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have default props', () => {
    expect(Navbar.defaultProps.menuIcon).toBeDefined();
    expect(Navbar.defaultProps.searchIcon).toBeDefined();
    expect(Navbar.defaultProps.dotsIcon).toBeDefined();
  });

  test('should render with navigationTitle', () => {
    const props = {
      ...props,
      navigationTitle: 'Custom Navigation Title',
    };

    const wrapper = shallow(<Navbar {...props} />);
    expect(wrapper.text()).toEqual('Custom Navigation Title');
  });
});
