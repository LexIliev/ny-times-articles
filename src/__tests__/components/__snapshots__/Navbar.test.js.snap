// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should match snapshot and render correctly 1`] = `
<header
  className="navbar"
>
  <img
    alt="Tree Horizontal Lines Icon"
    className="navbar__menu"
    src="/assets/three-horizontal-lines.svg"
  />
  <h2
    className="navbar__title"
  >
    Navigation title
  </h2>
  <div
    className="navbar__icons-wrapper"
  >
    <img
      alt="Search Icon"
      className="navbar__item"
      src="/assets/search.svg"
    />
    <img
      alt="Dots Icon"
      className="navbar__item"
      src="/assets/dots.svg"
    />
  </div>
</header>
`;
